templates/cc/repeated.go

```diff
@@ -27,22 +27,22 @@ const repTpl = `
 
 	{{ if $r.GetUnique }}
 	// Implement comparison for wrapped reference types
-	struct cmp {
+	struct {{ $safeTyp }}_cmp {
```

Comment:
This seems a bit ugly honestly. How about instead, something like:

```cpp
    struct cmp_{{ $f.Descriptor.Number }} {
```

Does that work, or am I maybe missing something important? This also avoids using double underscores, which are technically globally reserved identifiers in C++ (not that this is of very great importance considering `protoc` will happily generate and use identifiers with double underscores in them.)